src/DataValues/BoundedQuantityValue.php

```diff
@@ -0,0 +1,381 @@
+<?php
+
+namespace DataValues;
+
+use InvalidArgumentException;
+
+/**
+ * Class representing a quantity with associated unit and uncertainty interval.
+ * The amount is stored as a @see DecimalValue object.
+ *
+ * For quantities with no known uncertainty interval, see QuantityValue.
+ * For simple numeric amounts use @see NumberValue.
+ *
+ * @since 0.1
+ *
+ * @license GPL-2.0+
+ * @author Daniel Kinzler
+ */
+class BoundedQuantityValue extends QuantityValue {
+
+	/**
+	 * The quantity's upper bound
+	 *
+	 * @var DecimalValue
+	 */
+	private $upperBound;
+
+	/**
+	 * The quantity's lower bound
+	 *
+	 * @var DecimalValue
+	 */
+	private $lowerBound;
+
+	/**
+	 * Constructs a new QuantityValue object, representing the given value.
+	 *
+	 * @since 0.1
+	 *
+	 * @param DecimalValue $amount
+	 * @param string $unit A unit identifier. Must not be empty, use "1" for unit-less quantities.
+	 * @param DecimalValue $upperBound The upper bound of the quantity, inclusive.
+	 * @param DecimalValue $lowerBound The lower bound of the quantity, inclusive.
+	 *
+	 * @throws IllegalValueException
+	 */
+	public function __construct( DecimalValue $amount, $unit, DecimalValue $upperBound, DecimalValue $lowerBound ) {
+		parent::__construct( $amount, $unit );
+
+		if ( $lowerBound->compare( $amount ) > 0 ) {
+			throw new IllegalValueException( '$lowerBound ' . $lowerBound->getValue() . ' must be <= $amount ' . $amount->getValue() );
+		}
+
+		if ( $upperBound->compare( $amount ) < 0 ) {
+			throw new IllegalValueException( '$upperBound ' . $upperBound->getValue() . ' must be >= $amount ' . $amount->getValue() );
+		}
+
+		$this->upperBound = $upperBound;
+		$this->lowerBound = $lowerBound;
+	}
+
+	/**
+	 * Returns a QuantityValue representing the given amount.
+	 * If no upper or lower bound is given, the amount is assumed to be absolutely exact,
+	 * that is, the amount itself will be used as the upper and lower bound.
+	 *
+	 * This is a convenience wrapper around the constructor that accepts native values
+	 * instead of DecimalValue objects.
+	 *
+	 * @note: if the amount or a bound is given as a string, the string must conform
+	 * to the rules defined by @see DecimalValue.
+	 *
+	 * @since 0.1
+	 *
+	 * @param string|int|float|DecimalValue|QuantityValue $number
+	 * @param string $unit A unit identifier. Must not be empty, use "1" for unit-less quantities.
+	 * @param string|int|float|DecimalValue|null $upperBound
+	 * @param string|int|float|DecimalValue|null $lowerBound
+	 *
+	 * @return self
+	 * @throws IllegalValueException
+	 */
+	public static function newFromNumber( $number, $unit = '1', $upperBound = null, $lowerBound = null ) {
+		if ( $number instanceof QuantityValue ) {
```

Comment:
This may already have a unit and bounds, which will be ignored. I don't think this is helpful.